### PR TITLE
Homepage with and without login

### LINKED ISSUE #8: Homepage with and without login
**As a User**
**I need** know, what to do, when I open the app the first time
**to be able to** properly start using the app

## Description

Image

## Acceptance Criteria
- [x] A welcome text is visible
- [x] two links direct the user to the projects and competitions
- [x] The user gets told, what he can do, when he signs in
- [x] a 'sign in' button is clickable
- [x] the 'sign in' button opens the google login API URL
- [x] after the successful authentication, the use is redirected to the 'complete profile' page

## Tasks
- [x] Use the Google API to authenticate the user
- [x] save the google-ID, name and password to the database
- [x] redirect the user to the 'complete profile' page

### PR DESCRIPTION
**Pull request by luederb**

### CODE REVIEW

File: backend/pom.xml [hunk 1]
```diff
@@ -44,6 +47,14 @@
...
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+        </dependency>
```
Comment (MarcellNF) [inline backend/pom.xml]: For what do you need this dependency? JUnit is integrated in Spring Boot

File: frontend/src/components/Home/Home.tsx [hunk 2]
```diff
@@ -1,9 +1,44 @@
...
+        familyName: "",
+        email: "",
+        picture: ""
+    });
+
+    function loadUser() {
+        axios.get('/api/users/me')
+            .then(response => {
+                setUserData(response.data)
+                console.log("User data loaded:", response.data);
```
Comment (MarcellNF) [inline frontend/src/components/Home/Home.tsx]: You should not use console.log in production code. Instead you can use a logger

File: frontend/src/components/Home/Home.tsx [hunk 3]
```diff
@@ -1,9 +1,44 @@
...
+            })
+            .catch((error) => {
+                console.error("An error occurred while loading user data:", error);
```
Comment (MarcellNF) [inline frontend/src/components/Home/Home.tsx]: You should not use console.log in production code. Instead you can use a logger

File: frontend/src/components/Home/HomeWithoutLogin/HomeWithoutLogin.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import './HomeWithoutLogin.css';
+import {Link} from "react-router-dom";
+
+
+export default function HomeWithoutLogin() {
+    function login() {
+        const host = window.location.host === 'localhost:5173' ? 'http://localhost:8080': window.location.origin
+        window.open(host + '/oauth2/authorization/google', '_self')
+        console.log("I want to sign in with google");
```
Comment (MarcellNF) [inline frontend/src/components/Home/HomeWithoutLogin/HomeWithoutLogin.tsx]: You should not use console.log in production code. Instead you can use a logger

**Review by MarcellNF — CHANGES_REQUESTED** [review]
So far so good, nice work! But I have some recommendations below

File: backend/pom.xml (hunk 1 above)
Comment (luederb) [reply backend/pom.xml]: Thanks! You're right, I don't need this. I deleted it

File: frontend/src/components/Home/Home.tsx (hunk 2 above)
Comment (luederb) [reply frontend/src/components/Home/Home.tsx]: Thanks! I overlooked these console.logs. I replaced them with a Logger

File: frontend/src/components/Home/Home.tsx (hunk 3 above)
Comment (luederb) [reply frontend/src/components/Home/Home.tsx]: Thanks! I overlooked these console.logs. I replaced them with a Logger

File: frontend/src/components/Home/HomeWithoutLogin/HomeWithoutLogin.tsx (hunk 4 above)
Comment (luederb) [reply frontend/src/components/Home/HomeWithoutLogin/HomeWithoutLogin.tsx]: Thanks! I overlooked these console.logs. I replaced them with a Logger

**Review by MarcellNF — APPROVED** [review]
LGTM!:)

**Pull request merged**